app/actions/automatic_topup_action.rb

```diff
@@ -8,12 +8,28 @@ class AutomaticTopupAction < ApplicationAction
   private
 
   def perform_implementation
-    transaction = Transaction.create!(
+    return if existing_transaction
```

Comment:
This code is supposed to be an ad-hoc script. it should not be in runtime as we suppose moving further the automatic top-up job will work correctly. 

We only should do this:
- fix the data in production so that it doesn't have invalid top-up record (manually or by an ad-hoc script)
- prevent creation of invalid top-up record
- write an ad-hoc script and run it in production that will create transactions for configs failed/not run today.